### ISSUE TITLE
[BUG] <title>peft的p-tuning_v2无法使用，维度没有对齐

### ISSUE THREAD

**Issue by ssgg-code** [QwenLM/Qwen#1016]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

```python
        if layer_past is not None:
            past_key, past_value = layer_past[0], layer_past[1]
            if self.use_cache_quantization:
                # use_cache_quantization:
                # present=((q_key,key_scale,key_zero_point),
                #          (q_value,value_scale,value_zero_point))
                key = (torch.cat((past_key[0], key[0]), dim=2),
                       torch.cat((past_key[1], key[1]), dim=2),
                       torch.cat((past_key[2], key[2]), dim=2))
                value = (torch.cat((past_value[0], value[0]), dim=2),
                         torch.cat((past_value[1], value[1]), dim=2),
                         torch.cat((past_value[2], value[2]), dim=2))
            else:
                # not use_cache_quantization:
                # present=(key,value)
                key = torch.cat((past_key, key), dim=1)
                value = torch.cat((past_value, value), dim=1)
```
`key = torch.cat((past_key, key), dim=1)`这个dim错误，应该是dim=2， peft的prefix encoder给的past_key_value的维度和Qwen代码中的维度关系没对应上，seq_len和head_num反了

### 期望行为 | Expected Behavior

期望能够正常使用peft的p tuning v2来微调qwen

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by jklj077** [comment]
In the recent Qwen1.5 release, its codebase has been integrated into the `transformers` package, aligning with the established practices of the `transformers` library. This integration signifies that the `transformers` ecosystem is now designed to natively support Qwen1.5 models in most scenarios without any additional configuration, including the `peft` library.

For more information about this integration, discussions on how to leverage Qwen1.5 within the transformers environment, and for updates on community feedback and enhancements, please visit the official Qwen1.5 GitHub repository at <https://github.com/QwenLM/Qwen1.5>.
